src/LibraryViewExtensionMSWebBrowser/Handlers/IconResourceProvider.cs

```diff
@@ -0,0 +1,280 @@
+        private string defaultIconName;
+        private DllResourceProvider embeddedDllResourceProvider;
+        //TODO remove these at some point in future after Dynamo 2.6 release.
+        private MethodInfo getForAssemblyMethodInfo;
+        private PropertyInfo LibraryCustomizationResourceAssemblyProperty;
+        private Type LibraryCustomizationType;
+
+        private LibraryViewCustomization customization;
+        //internal cache of url to base64 encoded image data. (url,tuple<data,extension>)
+        private Dictionary<string, (string,string)> urlToBase64DataCache = new Dictionary<string, (string, string)>();
+
+        /// <summary>
+        /// Default constructor for the IconResourceProvider
+        /// </summary>
+        /// <param name="pathManager">Path manager instance to resolve the 
+        /// customization resource path</param>
+        /// <param name="defaultIcon">Name of the default icon (including 
+        /// extension) to be used when it can't find the requested icon</param>
+        public IconResourceProvider(IPathManager pathManager, string defaultIcon = "default-icon.svg") : base(true)
+        {
+            this.pathManager = pathManager;
+            defaultIconName = defaultIcon;
+            var dynCore = typeof(DynamoModel).Assembly;
+            this.getForAssemblyMethodInfo = dynCore.GetType("Dynamo.Engine.LibraryCustomizationServices").GetMethod("GetForAssembly", BindingFlags.Static | BindingFlags.Public);
+            this.LibraryCustomizationType = dynCore.GetType("Dynamo.Engine.LibraryCustomization");
+            this.LibraryCustomizationResourceAssemblyProperty = LibraryCustomizationType.GetProperty("ResourceAssembly", BindingFlags.Instance|BindingFlags.Public);
+        }
+
+        /// <summary>
+        /// Additional constructor used to access customization resources and dll resource provider directly during icon lookup.
+        /// </summary>
+        /// <param name="pathManager"></param>
+        /// <param name="dllResourceProvider"></param>
+        /// <param name="customization"></param>
+        /// <param name="defaultIcon"></param>
+        public IconResourceProvider(IPathManager pathManager, DllResourceProvider dllResourceProvider,LibraryViewCustomization customization, string defaultIcon = "default-icon.svg") :
+            this(pathManager, defaultIcon)
+        {
+            this.customization = customization;
+            this.embeddedDllResourceProvider = dllResourceProvider;
+        }
+
+
+        /// <summary>
+        /// Retrieves the resource for a given url as a base64 encoded string.
+        /// ie data:image/png;base64, stringxxxyyyzzz
+        /// </summary>
+        /// <param name="url">url for the requested icon</param>
+        /// <param name="extension">Returns the extension to describe the type of resource.</param>
+        /// <returns></returns>
+        public string GetResourceAsString(string url, out string extension)
+        {
```

Comment:
this a new method and should be reviewed - here is where most of the work is done for looking up a url and finding the relevant icon.